src/components/activity-history/tests/activity-history-integration.test.ts

```diff
@@ -0,0 +1,306 @@
+import request from "supertest";
+import { describe } from "mocha";
+import { sinon, expect } from "../../../../test/utils/test-utils";
+import { testComponent } from "../../../../test/utils/helpers";
+import * as cheerio from "cheerio";
+import * as nock from "nock";
+import decache from "decache";
+import { PATH_DATA } from "../../../app.constants";
+
+const { url } = PATH_DATA.SIGN_IN_HISTORY;
+
+const DEFAULT_USER_SESSION = {
+  email: "test@test.com",
+  isPhoneNumberVerified: true,
+  isAuthenticated: true,
+  subjectId: "asdf",
+  state: {},
+  tokens: {
+    accessToken: "token",
+    idToken: "Idtoken",
+    refreshToken: "token",
+  },
+};
+
+describe("Integration:: Activity history", () => {
+  beforeEach(() => {
+    nock.cleanAll();
+  });
+
+  it("should return sign in history page", async () => {
+    const app = await appWithMiddlewareSetup();
+    await request(app)
+      .get(url)
+      .expect(function (res) {
+        const $ = cheerio.load(res.text);
+        expect(res.status).to.equal(200);
+        expect($(testComponent("activity-history")).length).to.equal(1);
+      });
+  });
+
+  it("should not return sign in history page if feature flag is off", async () => {
+    const app = await appWithMiddlewareSetup([], { hideActivityLog: true });
+    await request(app)
+      .get(url)
+      .expect(function (res) {
+        expect(res.status).to.equal(404);
+      });
+  });
+
+  it("should display without pagination when data is less than activityLogItemsPerPage", async () => {
+    const dataShort = [
+      {
+        event_type: "signed-in",
+        session_id: "asdf",
+        user_id: "string",
+        timestamp: "1689210000",
+        activities: [
```

Comment:
I think these tests should reflect the new activity_log table structure

Reply:
If it's too much hassle, perhaps we can wait after we merge https://github.com/govuk-one-login/di-account-management-frontend/pull/1172 or perhaps borrow this approach? whichever is easier https://github.com/govuk-one-login/di-account-management-frontend/pull/1172/files#diff-ef3df6a5514d9bc94baf4d3f411b0c9d6efc7d0aa16d1a6d54496afa90399237R85-R101

Reply:
I've changed the integration tests so they reflect the correct structure of the log and fixed a bug where empty rows could have been displayed on the activity history.